code/game/objects/items/weapons/explosives.dm

```diff
@@ -37,39 +54,57 @@
 		return ..()
 
 /obj/item/plastique/attack_self(mob/user as mob)
-	var/newtime = input(usr, "Please set the timer.", "Timer", 10) as num
 	if(user.get_active_hand() == src)
-		newtime = clamp(newtime, 10, 60000)
-		timer = newtime
-		to_chat(user, SPAN_NOTICE("Timer set for [timer] seconds."))
+		timer = setTimer(user)
+		if(timer)
+			to_chat(user, SPAN_NOTICE("Timer set for [DisplayTimeText(timer)] seconds."))
+
+/obj/item/plastique/proc/setTimer(mob/user as mob)
+	var/newtime = tgui_input_number(
+		user = user,
+		message = "Please set the timer (in seconds).",
+		title = "Timer",
+		default = 30, // 30 second default
+		max_value = 1200, // 20 minute maximum
```

Comment:
i think it's a good time to change this, 20 minute maximum is ridiculous anyway. should probably be 5 mins